adcc/adc_pp/state2state_transition_dm.py

```diff
@@ -20,12 +20,119 @@
 ## along with adcc. If not, see <http://www.gnu.org/licenses/>.
 ##
 ## ---------------------------------------------------------------------
+from adcc import block as b
 from adcc.AdcMethod import AdcMethod
+from adcc.functions import einsum
 from adcc.AmplitudeVector import AmplitudeVector
+from adcc.OneParticleOperator import OneParticleOperator
 
 import libadcc
 
-DISPATCH = {}  # None implemented
+
+def s2s_tdm_adc0(mp, amplitude_l, amplitude_r, intermediates):
+    x1 = amplitude_l["s"]
+    y1 = amplitude_r["s"]
+
+    dm = OneParticleOperator(mp, is_symmetric=False)
+    dm[b.oo] = -einsum('ja,ia->ij', x1, y1)
+    dm[b.vv] = einsum('ia,ib->ab', x1, y1)
+    return dm
+
+
+def s2s_tdm_adc2(mp, amplitude_l, amplitude_r, intermediates):
+    dm = s2s_tdm_adc0(mp, amplitude_l, amplitude_r, intermediates)
+
+    x1 = amplitude_l["s"]
+    y1 = amplitude_r["s"]
```

Comment:
Not sure about x and y. The rest of the code uses u1 and u2. Why not stick to u1 and x1 maybe?

Reply:
Done.